invoices/invoice_expiry_watcher.go

```diff
@@ -115,6 +176,31 @@ func makeInvoiceExpiry(paymentHash lntypes.Hash,
 	case channeldb.ContractOpen:
 		return makeTimestampExpiry(paymentHash, invoice)
 
+	// If an invoice has active htlcs, we want to expire it based on block
+	// height. We only do this for hodl invoices, since regular invoices
+	// should resolve themselves automatically.
+	case channeldb.ContractAccepted:
+		if !invoice.HodlInvoice {
+			log.Debugf("Invoice in accepted state not added to "+
+				"expiry watcher: %v", paymentHash)
+
+			return nil
+		}
+
+		var expiryHeights []uint32
+
+		for _, htlc := range invoice.Htlcs {
+			// We only care about accepted htlcs, since they will
+			// trigger force-closes.
+			if htlc.State != channeldb.HtlcStateAccepted {
+				continue
+			}
+
+			expiryHeights = append(expiryHeights, htlc.Expiry)
```

Comment:
We do call `makeHeightExpiry` twice (`makeInvoiceExpiry`/`addHtlcs`) and the `addHtlcs` is called from outside of the context of the expiry watcher, so I thought it would be better to leave that logic in the watcher. Opinion not tightly held, so happy to do either :)

Reply:
IMHO reducing the number of allocations is worth it. Finding the minimum and passing it to `makeHeightExpiry` is probably the best solution since pre-allocating `expiryHeights` would allocate more than needed.

Reply:
Or we could pass `invoice` to `makeHeightExpiry` to avoid going through the slice a second time.

Reply:
We don't always have the `invoice` struct when we call `makeHeightExpiry`, so moving to just pass in minimum expiry height.